File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1, maxLen = 0;
        
        while(left <= right) {
            maxLen = max(maxLen, min(height[right], height[left]) * (right - left));
            if(height[left] < height[right])
                left++;
            else 
                right--;
        }
        
        return maxLen;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        
        int amountOfWater = (right - left) * Math.min(height[right], height[left]);
        while(left <= right) {
            amountOfWater = Math.max(amountOfWater, (right - left) * Math.min(height[right], height[left]));
            if(height[left] < height[right])
                left++;
            else right--;
        }
        
        return amountOfWater;
    }
}